From 9a7efe890445f2036bffb1c45f6f945d472bada8 Mon Sep 17 00:00:00 2001
From: Jukka Rissanen <jukka.rissanen@linux.intel.com>
Date: Thu, 19 Apr 2012 15:29:53 +0300
Subject: [PATCH] service: Changing default service was not correctly checked

Let the default_changed() function check whether the default service is
really changed as it is doing it correctly every time.

The extra check in the __connman_service_indicate_default() function is
causing problems if the user is moving services around. Then we might be
using the old DNS service data because DNS code did not get information
that default service was changed.

Fixes BMC#25027
---
 src/service.c |    7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/src/service.c b/src/service.c
index a0e8a11..d14431a 100644
--- a/src/service.c
+++ b/src/service.c
@@ -4662,12 +4662,7 @@ int __connman_service_clear_error(struct connman_service *service)
 
 int __connman_service_indicate_default(struct connman_service *service)
 {
-	struct connman_service *current = __connman_service_get_default();
-
-	DBG("service %p default %p", service, current);
-
-	if (current == service)
-		return 0;
+	DBG("service %p", service);
 
 	default_changed();
 
-- 
1.7.10

